nixos/modules/services/networking/keepalived/default.nix

```diff
@@ -318,27 +318,32 @@ in
     };
 
     systemd.services.keepalived = let
-      finalConfigFile = if cfg.secretFile == null then keepalivedConf else "/run/keepalived/keepalived.conf";
+      finalConfigFile = "/run/keepalived/keepalived.conf";
+      populateSecrets =  ''
+        umask 077
+        ${pkgs.envsubst}/bin/envsubst -i "${keepalivedConf}" > ${finalConfigFile}
+      '';
     in {
       description = "Keepalive Daemon (LVS and VRRP)";
       after = [ "network.target" "network-online.target" "syslog.target" ];
       wants = [ "network-online.target" ];
+      reloadIfChanged = true;
+      restartTriggers = [ pkgs.keepalived cfg.snmp.enable ];
```

Comment:
The configuration file in general is also target for reloading no?

Reply:
I'm a bit confused by the behavior of `reloadIfChanged` together with `restartTriggers`. My understanding was, that `reloadIfChanged` would reload the service instead of restarting it (the default), when any part of the service changes. `restartTriggers` then provide a way to override this behavior and force a restart nonetheless, if the contents of restartTriggers changes.

This doesn't seem to be the case however, since the restart test that I added also does a reload and thus fails.